promql/testdata/functions.test

```diff
@@ -341,19 +341,54 @@ clear
 
 # Test for increase()/xincrease()/yincrease() with counter reset.
 # When the counter is reset, it always starts at 0.
-# So the sequence 1003 1002 (decreasing counter = reset) is interpreted the same as 1003 1000 1001 1002.
-# Prometheus assumes it missed the intermediate values 1000 and 1001.
+# So the sequence 1003 2 (decreasing counter = reset) is interpreted the same as 1003 0 1 2.
+# Prometheus assumes it missed the intermediate values 0 and 1.
 load 5m
-	http_requests{path="/foo"}	1000 1001 1002 1003 1002 1003 1004
+	http_requests{path="/foo"}	1000 1001 1003 1006 4 9 16
```

Comment:
I made each gap different so there's no chance of getting correct results if the code were to have an off-by-1 bug about what sample it looked at.